pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -107,9 +107,11 @@ def __init__(self, filepath: Optional[str] = None, monitor: str = 'val_loss', ve
         self.monitor = monitor
         self.verbose = verbose
         if filepath is None:  # will be determined by trainer at runtime
-            self.dirpath, self.filename = None, None
+            self.dirpath, self.filename = None, filename
         else:
-            if os.path.isdir(filepath):
+            if filename is not None:
```

Comment:
```suggestion
            if filename:
```
an empty string is also not good